### PR TITLE
Launch CRIBs on pre-release tags

### PR DESCRIPTION
**Pull request by chainchad**
- Refactor to use composite action
- Fix bug to get only tagged image
- Launch CRIB on pre-release tags

### CODE REVIEW

File: .github/workflows/build-publish.yml [hunk 1]
```diff
@@ -36,6 +38,9 @@ jobs:
             echo "git-tag-type=core" | tee -a "$GITHUB_OUTPUT"
             echo "ecr-image-name=chainlink/chainlink" | tee -a "$GITHUB_OUTPUT"
           fi
+      - name: Check release tag
+        id: release-tag-check
+        uses: smartcontractkit/.github/actions/release-tag-check@re-2954/migrate-release-tag-check-action # TODO: set version/commit when released
```
Comment (chainchad) [inline .github/workflows/build-publish.yml]: TODO

File: .github/workflows/crib-integration-test.yml [hunk 2]
```diff
@@ -20,101 +22,22 @@ jobs:
...
-          K8S_STAGING_INGRESS_SUFFIX: ${{ secrets.K8S_STAGING_INGRESS_SUFFIX }}
-          CRIB_NAMESPACE: ${{ steps.deploy-crib.outputs.devspace-namespace }}
-          CRIB_NETWORK: geth
-          CRIB_NODES: 5
-          GAP_URL: ${{ secrets.GAP_URL }}
-          SETH_LOG_LEVEL: info
-          # RESTY_DEBUG: true
-          TEST_PERSISTENCE: true
-          E2E_TEST_CHAINLINK_IMAGE: public.ecr.aws/chainlink/chainlink
-          E2E_TEST_CHAINLINK_VERSION: latest
```
Comment (chainchad) [inline .github/workflows/crib-integration-test.yml:116]: Confirmed with @skudasov that we should not need these env vars in CRIB.

File: .github/actions/crib/action.yml [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+    required: true
+  aws-account-id:
+    description: "AWS Account ID"
+    required: true
+  api-gw-host-crib:
+    description: "API Gateway Host for CRIB"
+    required: true
+  api-gw-host-k8s:
+    description: "API Gateway Host for K8s"
+    required: true
```
Comment (chainchad) [inline .github/actions/crib/action.yml:23]: TODO: remove any refs to api-gw (old inputs).

File: .github/actions/crib/action.yml [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    - uses: cachix/install-nix-action@ba0dd844c9180cbf77aa72a116d6fbc515d0e87b # v27
+      with:
+        nix_path: nixpkgs=channel:nixos-unstable
+
+    - name: setup-gap crib
+      uses: smartcontractkit/.github/actions/setup-gap@7db14549894156db3c395e30b49f9d00754185ff # setup-gap@3.0.1
+      with:
+        aws-role-duration-seconds: ${{ inputs.aws-role-duration-seconds }}
+        aws-role-arn: ${{ inputs.aws-role-arn }}
+        api-gateway-host: ${{ inputs.api-gw-host-crib }}
```
Comment (chainchad) [inline .github/actions/crib/action.yml:80]: TODO

File: .github/actions/crib/action.yml [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        use-private-ecr-registry: true
+        use-tls: true
+        proxy-port: 8080
+
+    - name: setup-gap k8s
+      uses: smartcontractkit/.github/actions/setup-gap@7db14549894156db3c395e30b49f9d00754185ff # setup-gap@3.0.1
+      with:
+        aws-role-duration-seconds: ${{ inputs.aws-role-duration-seconds }}
+        aws-role-arn: ${{ inputs.aws-role-arn }}
+        api-gateway-host: ${{ inputs.api-gw-host-k8s }}
```
Comment (chainchad) [inline .github/actions/crib/action.yml:94]: TODO

File: .github/workflows/crib-integration-test.yml [hunk 6]
```diff
@@ -20,101 +22,22 @@ jobs:
...
+        uses: ./.github/actions/crib
         with:
-          aws-role-duration-seconds: 3600 # 1 hour
           aws-role-arn: ${{ secrets.AWS_OIDC_CRIB_ROLE_ARN_STAGE }}
-          api-gateway-host: ${{ secrets.AWS_API_GW_HOST_CRIB_STAGE }}
           aws-region: ${{ secrets.AWS_REGION }}
-          ecr-private-registry: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
+          aws-account-id: ${{ secrets.AWS_ACCOUNT_ID_PROD }}
+          api-gw-host-crib: ${{ secrets.AWS_API_GW_HOST_CRIB_STAGE }}
+          api-gw-host-k8s: ${{ secrets.AWS_API_GW_HOST_K8S_STAGE }}
```
Comment (chainchad) [inline .github/workflows/crib-integration-test.yml:35]: TODO

Comment (chainchad) [reply .github/workflows/crib-integration-test.yml:35]: Going to convert `setup-gap` to use GAP v2 separately.

File: .github/actions/crib/action.yml [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+        CRIB_NETWORK: geth
+        CRIB_NODES: 5
+        GAP_URL: ${{ inputs.api-gw-host-crib }}
+        SETH_LOG_LEVEL: info
+        TEST_PERSISTENCE: true
+      run: go test -v -run TestCRIBChaos
+
+    - name: Destroy CRIB Environment
+      if: inputs.crib-destroy-environment == 'true' && always() && steps.deploy-crib.outputs.devspace-namespace != ''
+      uses: smartcontractkit/.github/actions/crib-purge-environment@c0b38e6c40d72d01b8d2f24f92623a2538b3dedb
```
Comment (chainchad) [inline .github/actions/crib/action.yml]: TODO: Add version

File: .github/actions/crib/action.yml [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+      id: token
+      uses: smartcontractkit/.github/actions/setup-github-token@c0b38e6c40d72d01b8d2f24f92623a2538b3dedb
+      with:
+        aws-role-arn: ${{ inputs.aws-token-issuer-role-arn }}
+        aws-lambda-url: ${{ inputs.aws-token-issuer-lambda-url }}
+        aws-region: ${{ inputs.aws-region }}
+        aws-role-duration-seconds: "1800"
+
+    - name: Deploy and validate CRIB Environment for Core
+      uses: smartcontractkit/.github/actions/crib-deploy-environment@815e0d550527897746e889441407926d7e28169c
```
Comment (chainchad) [inline .github/actions/crib/action.yml]: TODO: Add version comment

File: .github/actions/crib/action.yml [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+        ecr-private-registry: ${{ inputs.aws-account-id }}
+        k8s-cluster-name: ${{ inputs.k8s-cluster-name }}
+        gap-name: k8s
+        use-private-ecr-registry: true
+        use-k8s: true
+        proxy-port: 8443
+
+    - name: Setup GitHub token using GATI
+      id: token
+      uses: smartcontractkit/.github/actions/setup-github-token@c0b38e6c40d72d01b8d2f24f92623a2538b3dedb
```
Comment (chainchad) [inline .github/actions/crib/action.yml]: TODO: Add version comment

File: .github/actions/crib/action.yml (hunk 5 above)
Comment (chainchad) [reply .github/actions/crib/action.yml:94]: Going to convert `setup-gap` to use GAP v2 separately.

File: .github/actions/crib/action.yml (hunk 4 above)
Comment (chainchad) [reply .github/actions/crib/action.yml:80]: Going to convert `setup-gap` to use GAP v2 separately.

File: .github/actions/crib/action.yml (hunk 3 above)
Comment (chainchad) [reply .github/actions/crib/action.yml:23]: Going to convert `setup-gap` to use GAP v2 separately.

File: .github/actions/crib/action.yml [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+      uses: smartcontractkit/.github/actions/crib-deploy-environment@815e0d550527897746e889441407926d7e28169c # crib-deploy-environment@7.4.0
+      id: deploy-crib
+      with:
+        github-token: ${{ steps.token.outputs.access-token }}
+        aws-ecr-private-registry: ${{ inputs.aws-account-id }}
+        aws-region: ${{ inputs.aws-region }}
+        aws-role-arn: ${{ inputs.aws-role-arn }}
+        ingress-base-domain: ${{ inputs.ingress-base-domain }}
+        k8s-api-endpoint: ${{ inputs.api-gw-host-k8s }}
+        k8s-cluster-name: ${{ inputs.k8s-cluster-name }}
```
Comment (chainchad) [inline .github/actions/crib/action.yml]: TODO

**Review by njegosrailic — DISMISSED** [review]

**Review by njegosrailic — APPROVED** [review]

File: .github/workflows/crib-integration-test.yml (hunk 2 above)
Comment (chainchad) [reply .github/workflows/crib-integration-test.yml:116]: Looks like we did end up needing this.

**Review by Bwest981 — APPROVED** [review]

**Pull request merged**